File: iOS/Utils/Utils/Utils/UIUtils.m
#import "UIUtils.h"

@implementation UIUtils

static UIColor * _defaultColor;

+(void) setDefaultColor:(UIColor *)color {
    _defaultColor = color;
}

+(UIColor *) getDefaultColor {
    if (!_defaultColor) {
        _defaultColor = [UIColor colorWithRed:0x00 green:0xA5 / 0xFF blue:0xED / 0xFF alpha:1];
    }
    return _defaultColor;
}

+(CGSize) getStatusBarSize {
    return [UIApplication sharedApplication].statusBarFrame.size;
}

+(CGSize) getScreenSize {
    return [UIScreen mainScreen].bounds.size;
}

+(void) setNavBar: (UINavigationBar *)nav bgColor: (UIColor *)bgColor textColor: (UIColor *)textColor {
    nav.barTintColor = bgColor;
    nav.tintColor = textColor;
    nav.titleTextAttributes = [NSDictionary dictionaryWithObjectsAndKeys:textColor, NSForegroundColorAttributeName, nil];
}

+(void) setTabBar: (UITabBar *)tab bgColor: (UIColor *)bgColor textColor: (UIColor *)textColor {
    tab.tintColor = textColor;
    tab.barTintColor = bgColor;
}

+(void) setStatusBarStyle: (BOOL)light {
    [[UIApplication sharedApplication] setStatusBarStyle:(light ? UIStatusBarStyleLightContent : UIStatusBarStyleDefault) animated:NO];
}

+(CGFloat) getStatusBarHeight {
    return [UIApplication sharedApplication].statusBarFrame.size.height;
}

+(CGSize) getAppFrameSize {
    return [UIScreen mainScreen].applicationFrame.size;
}

+(CGFloat) getNavigationBarHeight: (UINavigationBar *)nav {
    return nav.frame.size.height;
}

+(TitleInfo) setTextTitleForSideMenu: (UIViewController *)controller nav: (UINavigationBar *)nav txt: (UILabel *)txt {
    CGFloat y = [self getStatusBarSize].height;
    CGFloat w = [self getScreenSize].width;
    CGFloat h = [self getNavigationBarHeight:nav];
    CGFloat t = y + h;
    CGFloat rh = [self getScreenSize].height - t;
    UIColor * backColor = [self getDefaultColor];
    txt.frame = CGRectMake(0, y, w, h);
    txt.backgroundColor = backColor;
    txt.textColor = [UIColor whiteColor];
    h = [self getStatusBarSize].height;
    UILabel * txtImmersion = [[UILabel alloc] initWithFrame: CGRectMake(0, 0, w, h)];
    txtImmersion.backgroundColor = backColor;
    [controller.view addSubview:txtImmersion];
    TitleInfo ret;
    ret.top = t;
    ret.remainHeight = rh;
    return ret;
}

+(void)makeTableViewFullSize:(UITableView *)table items:(NSArray *)items {
    NSInteger height = 0;
    for (int i=0; i<items.count; i++) {
        CGRect rect = [table rectForRowAtIndexPath:[NSIndexPath indexPathForRow:i inSection:0]];
        height += rect.size.height;
    }
    CGRect frame = [table frame];
    frame.size.height = height;
    [table setFrame:frame];
}

+(void)setViewWidthPercent:(UIView *)view percent:(CGFloat)percent {
    CGSize size = [self getScreenSize];
    CGRect frame = [view frame];
    frame.size.width = size.width * percent;
    [view setFrame:frame];
}

+(void)setViewHeightPercent:(UIView *)view percent:(CGFloat)percent {
    CGSize size = [self getScreenSize];
    CGRect frame = [view frame];
    frame.size.height = size.height * percent;
    [view setFrame:frame];
}

+(void)setViewSizePercent:(UIView *)view percent:(CGFloat)percent {
    CGSize size = [self getScreenSize];
    CGRect frame = [view frame];
    frame.size.width = size.width * percent;
    frame.size.height = size.height * percent;
    [view setFrame:frame];
}

@end

File: Swift/Utils/UIUtils.swift
//
//  UIUtils.swift
//  YibanJob
//
//  Created by rarnu on 1/10/15.
//  Copyright (c) 2015 yiban. All rights reserved.
//

import UIKit

class UIUtils: NSObject {
    
    class func getDefaultColor() -> UIColor {
        return UIColor(red: 0x00, green: 0xA5 / 0xFF, blue: 0xED / 0xFF, alpha: 1)
    }
    
    class func getStatusBarSize() -> CGSize {
        return UIApplication.sharedApplication().statusBarFrame.size
    }
    
    class func getScreenSize() -> CGSize {
        return UIScreen.mainScreen().bounds.size
    }
    
    class func setNavBar(nav: UINavigationBar, bgColor: UIColor, textColor: UIColor) {
        nav.barTintColor = bgColor
        nav.tintColor = textColor
        nav.titleTextAttributes = NSDictionary(objectsAndKeys: textColor, NSForegroundColorAttributeName)
    }
    
    class func setTabBar(tab: UITabBar, bgColor: UIColor, textColor: UIColor) {
        tab.tintColor = textColor
        tab.barTintColor = bgColor
    }
    
    class func setStatusBarStyle(light: Bool) {
        UIApplication.sharedApplication().setStatusBarStyle(light ? UIStatusBarStyle.LightContent : UIStatusBarStyle.Default, animated: false)
    }

    class func setTextTitleForSideMenu(controller: UIViewController, txt: UILabel) -> (CGFloat, CGFloat) {
        var y = UIUtils.getStatusBarSize().height
        var w = UIUtils.getScreenSize().width
        var h: CGFloat = getStatusBarHeight()
        var t: CGFloat = y + h
        var rh: CGFloat = UIUtils.getScreenSize().height - t
        var backColor = getDefaultColor()
        txt.frame = CGRectMake(0, y, w, h)
        txt.backgroundColor = backColor
        txt.textColor = UIColor.whiteColor()
        h = UIUtils.getStatusBarSize().height
        var txtImmersion = UILabel(frame: CGRectMake(0, 0, w, h))
        txtImmersion.backgroundColor = backColor
        controller.view.addSubview(txtImmersion)
        return (t, rh)
    }
    
    class func getStatusBarHeight() -> CGFloat {
        return UIApplication.sharedApplication().statusBarFrame.height
    }
    
    class func getAppFrameSize() -> CGSize {
        return UIScreen.mainScreen().applicationFrame.size
    }
}
